src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,129 @@
+            ResultSet resultSet = statement.executeQuery();
+            Book book = null;
+            if (resultSet.next()) {
+                book = getBookFromResultSet(resultSet);
+            }
+            return Optional.ofNullable(book);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book from id " + id + ":", e);
+        }
+    }
+
+    @Override
+    public List<Book> getAll() {
+        List<Book> books = new ArrayList<>();
+        String getAllBookQuery = "SELECT * FROM books WHERE is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(getAllBookQuery)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                books.add(getBookFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all books from DB:", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String updateBookQuery =
+                "UPDATE books SET title = ?, price = ? WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(updateBookQuery)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Excepted to update at least one row, "
+                        + "but updated 0 rows:",
+                        new RuntimeException());
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book by id:" + book.getId(), e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteBookQuery = "UPDATE books SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                            .prepareStatement(deleteBookQuery)) {
+            statement.setLong(1, id);
+            int affectedRows = statement.executeUpdate();
+            return affectedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete book by id:" + id + ":", e);
+        }
+    }
+
+    private Book getBookFromResultSet(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
+        String title = resultSet.getString("title");
+        BigDecimal price = resultSet.getBigDecimal("price");
```

Comment:
you can save strings to constants

Reply:
Did you mean to make these fields final?

Reply:
do you mean input parameters in methods?